### PR TITLE
#6 Creat MessageController & UserController based on the current looks o…

### PR DESCRIPTION
**Pull request by ajohnsson95**
…f the service classes.

### Related Issue(s)
Issue #6 

### Proposed Changes
Creat Controller class for Message & User

### Description
Controller classes created based on current Serviceclass.

### CODE REVIEW

**Comment by ajohnsson95** [comment]
Controller classes will fail due to missing MessageService & UserService, will make a new commit once the service classes have been merged

File: src/main/java/dev/ebrydeu/spring_boot_library/controllers/MessageController.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import dev.ebrydeu.spring_boot_library.domain.dto.MessageDto;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/Messages")
```
Comment (Ebyrdeu) [inline src/main/java/dev/ebrydeu/spring_boot_library/controllers/MessageController.java:11]: use lowercase

File: src/main/java/dev/ebrydeu/spring_boot_library/controllers/MessageController.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    @GetMapping
+    public List<MessageDto> getAllMessages(){
+        return messageService.findAllMessages();
+    }
+
+    @GetMapping("/author/{author}")
+    public List<MessageDto> getMessagesByAuthor(@PathVariable String author){
+        return messageService.findMessagesByAuthor(author);
+    }
```
Comment (Ebyrdeu) [inline src/main/java/dev/ebrydeu/spring_boot_library/controllers/MessageController.java:24]: #20  Now, That I look Att it what exactly String author supposed to be?  Is it Id name or just POJO? Does not make sense for me as for now.

File: src/main/java/dev/ebrydeu/spring_boot_library/controllers/UserController.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @GetMapping("{id}")
+    public UserDto getUserById(@PathVariable Long id) {
+        try {
+            return userService.getUserById(id);
+        } catch (Exception e) {
+            return null;
+      }
+    }
```
Comment (Ebyrdeu) [inline src/main/java/dev/ebrydeu/spring_boot_library/controllers/UserController.java:28]: We do have null check on UserService I believe ? So, it going to throw a notFoundExpection error for user if cannot find it. (But need to be tested when service layer is done). 

```suggestion
    @GetMapping("{id}")
    public UserDto getUserById(@PathVariable Long id) {
            return userService.getUserById(id);
    }

**Review by Ebyrdeu — COMMENTED** [review]
Just going to comment as you said you're not done with it

**Review by ValentinaSukonina — COMMENTED** [review]
You might consider adding input validation annotations such as @Valid and @NotNull to ensure that the data sent to the endpoints is valid.
@PostMapping
@ResponseStatus(HttpStatus.CREATED)
public UserDto createUser(@Valid @NotNull @RequestBody UserDto userDto) {
        return userService.createUser(userDto);
    }
    That also mean our Dto classes has to be annotated accordingly (Annotations on Entity level only applicable to the database, not our application)

**Comment by ajohnsson95** [comment]
> You might consider adding input validation annotations such as @Valid and @NotNull to ensure that the data sent to the endpoints is valid. @PostMapping @ResponseStatus(HttpStatus.CREATED) public UserDto createUser(@Valid @NotNull @requestbody UserDto userDto) { return userService.createUser(userDto); } That also mean our Dto classes has to be annotated accordingly (Annotations on Entity level only applicable to the database, not our application)

sounds good I belive the easiest way to do this would be to add some anotations in the DTO class 
for example 
    @NotBlank(message = "First name is required")
    @Size(max = 255, message = "First name must be less than 256 characters")
    private String firstName;
 
 but that would probably force us too change the UserDto to a class and not a record if im correct

File: src/main/java/dev/ebrydeu/spring_boot_library/controllers/MessageController.java (hunk 2 above)
Comment (ajohnsson95) [reply src/main/java/dev/ebrydeu/spring_boot_library/controllers/MessageController.java:24]: Should we consider just skipping author all together, or is the one who created the message also the author?

**Comment by ajohnsson95** [comment]
@Ebyrdeu @ValentinaSukonina 
Still not added the @Validated annotation should not be to hard to do but i belive changes have to be done in the Dto records, to set up validations there aswell, and possibly add the 
               
               <dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-validation</artifactId>
		</dependency>
		
to our pom.xml

**Comment by ValentinaSukonina** [comment]
> @Ebyrdeu @ValentinaSukonina Still not added the @validated annotation should not be to hard to do but i belive changes have to be done in the Dto records, to set up validations there as well, and possibly add the
> 
> ```
>            <dependency>
> 		<groupId>org.springframework.boot</groupId>
> 		<artifactId>spring-boot-starter-validation</artifactId>
> 	</dependency>
> ```
> 
> to our pom.xml

You are right, it should be implemented on Dto level. Check this https://stackoverflow.com/questions/64186922/java-records-validation-anotation

File: src/main/java/dev/ebrydeu/spring_boot_library/controllers/MessageController.java (hunk 2 above)
Comment (Ebyrdeu) [reply src/main/java/dev/ebrydeu/spring_boot_library/controllers/MessageController.java]: You can skip for now. We will se how it will look without it

Comment (ValentinaSukonina) [reply src/main/java/dev/ebrydeu/spring_boot_library/controllers/MessageController.java]: I think you are right, my understanding the author is the one who created message, meaning we should be able to access this from UserEntity by id, and therefore do not need this in MessageEntity

**Review by ValentinaSukonina — APPROVED** [review]
Good work

**Review by Ebyrdeu — APPROVED** [review]

**Pull request merged**